pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/presence/HereNowApp.java

```diff
@@ -21,29 +21,61 @@ public static void main(String[] args) throws PubNubException {
 
         PubNub pubnub = PubNub.create(configBuilder.build());
 
-        // Get presence information for specified channels
+        // Get presence information for specified channels with pagination support
         pubnub.hereNow()
                 .channels(Arrays.asList("coolChannel", "coolChannel2"))
                 .includeUUIDs(true)
+                .limit(100)
                 .async(result -> {
                     result.onSuccess((PNHereNowResult res) -> {
+                        printHereNowResult(res);
 
-                        System.out.println("Total Channels: " + res.getTotalChannels());
-                        System.out.println("Total Occupancy: " + res.getTotalOccupancy());
-
-                        for (PNHereNowChannelData channelData : res.getChannels().values()) {
-                            System.out.println("---");
-                            System.out.println("Channel: " + channelData.getChannelName());
-                            System.out.println("Occupancy: " + channelData.getOccupancy());
-                            System.out.println("Occupants:");
-                            for (PNHereNowOccupantData occupant : channelData.getOccupants()) {
-                                System.out.println("UUID: " + occupant.getUuid() + " State: " + occupant.getState());
-                            }
+                        // Check if more results are available
+                        if (res.getNextOffset() != null) {
+                            System.out.println("\nMore results available. Fetching next page...\n");
+
+                            // Fetch next page using the offset from previous response
+                            pubnub.hereNow()
+                                    .channels(Arrays.asList("coolChannel", "coolChannel2"))
+                                    .includeUUIDs(true)
+                                    .limit(100)
+                                    .offset(res.getNextOffset())
+                                    .async(result2 -> {
+                                        result2.onSuccess((PNHereNowResult res2) -> {
+                                            printHereNowResult(res2);
+
+                                            // Continue pagination if needed by checking res2.getNextOffset()
+                                        }).onFailure((PubNubException exception) -> {
+                                            System.out.println("Error retrieving hereNow data: " + exception.getMessage());
+                                        });
+                                    });
```

Comment:
Will it make sense / possible to create local closure which will be called from initial and nested `hereNow` calls, but in addition to `result` will pass `offset` for example and inside if it is not `nil`, make another `hereNow` calls - it will give user better idea because novice may think that it is ok to nest `hereNow` calls as long as possible.